packages/atlas/src/providers/nftActions/hooks.ts

```diff
@@ -0,0 +1,49 @@
+import { useCallback, useContext } from 'react'
+
+import { NftActionsContext } from './provider'
+
+export const useNftActions = () => {
+  const ctx = useContext(NftActionsContext)
+  if (ctx === undefined) {
+    throw new Error('useNftActions must be used within NftActionsProvider')
+  }
+
+  const { currentAction, currentNftId, setCurrentAction, setCurrentNftId } = ctx
+
+  const openNftPurchase = useCallback(
+    (nftId: string) => {
+      setCurrentNftId(nftId)
+      setCurrentAction('purchase')
+    },
+    [setCurrentAction, setCurrentNftId]
+  )
+
+  const openNftPutOnSale = useCallback(
+    (nftId: string) => {
+      setCurrentNftId(nftId)
+      setCurrentAction('putOnSale')
+    },
+    [setCurrentAction, setCurrentNftId]
+  )
+
+  const openNftSettlement = useCallback(
+    (nftId: string) => {
+      setCurrentNftId(nftId)
+      setCurrentAction('settle')
+    },
+    [setCurrentAction, setCurrentNftId]
+  )
```

Comment:
We're creating three functions that basically do the same. Is there any reason why we do that? Can we make just one and pass, `settle, purchase` or `putOnSale` as a second argument?

Reply:
Don't know, just thought that maybe in the future we may want to pass separate params to different types of actions. I guess it doesn't hurt us to have them separate